src/plugins/intel_cpu/src/transformations/snippets/x64/pass/fuse_brgemm_cpu_postops.cpp

```diff
@@ -0,0 +1,434 @@
+}
+
+auto brgemm_predicate = [](const Output<Node>& output) {
+    const auto brgemm = output.get_node_shared_ptr();
+    // Note: postops are not supported in case of blocking enabled,
+    // so f32 precision is not included in supported list
+    // Ticket: 165567
+    static const ov::element::TypeVector supported_in_precisions{ov::element::bf16, ov::element::i8, ov::element::u8};
+    return has_static_rank()(output) && consumers_count(1)(output) &&
+           type_matches_any(supported_in_precisions)(brgemm->input_value(0)) &&
+           type_matches_any(supported_in_precisions)(brgemm->input_value(1));
+};
+
+auto scalar_predicate = [](const Output<Node>& output) {
+    return type_matches(ov::element::f32)(output);
+};
+
+}  // namespace
+
+pass::FuseBrgemmCPUPostops::FuseBrgemmCPUPostops(std::set<size_t>& brgemm_external_params_idces)
+    : m_brgemm_external_params_idces(brgemm_external_params_idces) {
+    add_matcher<FuseConvert>();
+    add_matcher<FuseUnaryEltwise>();
+    add_matcher<FuseScaleShift>();
+    add_matcher<FuseClip>();
+    add_matcher<FuseScalarEltwise>();
+    add_matcher<FuseBinaryEltwise>(m_external_params);
+}
+
+bool pass::FuseBrgemmCPUPostops::run_on_model(const std::shared_ptr<ov::Model>& m) {
+    RUN_ON_MODEL_SCOPE(FuseBrgemmCPUPostops);
+    OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "ov::intel_cpu::pass::FuseBrgemmCPUPostops")
+    const auto res = GraphRewrite::run_on_model(m);
+    for (const auto& param : m_external_params) {
+        m_brgemm_external_params_idces.insert(m->get_parameter_index(param));
+    }
+    return res;
+}
+
+pass::FuseConvert::FuseConvert() {
+    MATCHER_SCOPE(FuseConvert);
+
+    auto m_brgemm = wrap_type<BrgemmCPU>(brgemm_predicate);
+    auto m_convert = wrap_type<ConvertSaturation>({m_brgemm});
+
+    auto callback = [=](Matcher& m) {
+        OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "ov::intel_cpu::pass::FuseConvert")
+        const auto& pattern_map = m.get_pattern_value_map();
+        const auto convert = pattern_map.at(m_convert).get_node_shared_ptr();
+        const auto brgemm = ov::as_type_ptr<BrgemmCPU>(pattern_map.at(m_brgemm).get_node_shared_ptr());
+
+        // Ticket 165567: In case of AMX, brgemm may require two kernels: main and tail.
+        // Intermediate results must be stored in the output buffer for accumulation with tail kernel results.
+        // Forcing an output precision with a smaller bit width for output buffer causes out-of-bounds memory writes
+        // during intermediate results storage, so the convert fusion is skipped in this case.
+        if (brgemm_utils::with_amx(brgemm->get_type())) {
+            const auto& cur_out_precision = brgemm->get_output_element_type(0);
+            const auto& new_out_precision = convert->get_output_element_type(0);
+            if (cur_out_precision.bitwidth() > new_out_precision.bitwidth()) {
+                return false;
+            }
+        }
```

Comment:
BTW if `K_dim % inner_k_blk == 0`, we can fuse `Convert` even in AMX scenario

- I remember you made some experiments on int8 SD. Could this additional check be useful for SD-case ?
- Also our target model is inferred on AMX platform. The target model has `FQ` after `Brgemm` - this `Brgemm` should fuse everything after itself - even `Convert i8`. Did you check that all postops were fused into `Brgemm`?

Reply:
I locally dumped the body after data flow, `ConvertSaturation` is on graph:
![image](https://github.com/user-attachments/assets/e37a6d28-2527-4b02-b611-47711b72f908)
But there shouldn't be any ops between `Brgemms`

Reply:
I relaxed the FuseConvert limitations: now the fusion doesn't happen only if internal k blocking is needed (`brgemm_utils::repacking::compute_inner_k_block` is used for such purpose)